### DISCUSSION TITLE
Can I pin the currently locked dependency version while building?

### DISCUSSION THREAD

**Discussion by Popkornium18** [python-poetry/poetry#8602] [Q&A] [answered]
I don't want to bump version contraints all the time, especially for dependencies that release new major versions or unstable minor versions very regularly.

I thought it would be a good solution to set all dependency constraints to `"*"` and only pin specific versions if tests fail. However, now the built packages will also always use the latest version of dependencies which is not what I want, because that would mean I will most likely realize that a new dependency version requires changes because production breaks and not when I run tests after an update.

Is there a way to let poetry upgrade to the latest version when doing a `poetry update` but pin versions when actually building the packages?

**Comment by dimbleby** [comment, marked as answer]
no

**Reply by Popkornium18** [reply]
Would a PR implementing such a feature be accepted?

**Reply by dimbleby** [reply]
#2778
